enhancements/cluster-logging/forward_to_cloudwatch.md

```diff
@@ -0,0 +1,226 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+[Amazon Cloudwatch][aws-cw] is a hosted monitoring and log storage service.
+This proposal extends the `ClusterLogForwarder` API with an output type for Cloudwatch.
+
+## Motivation
+
+Amazon Cloudwatch is a popular log store.
+We have requests from external and  Red Hat-internal customers to support it.
+
+### Goals
+
+Enable log forwarding to Cloudwatch.
+
+### Non-Goals
+
+Enable Cloudwatch metric collection.
+
+## Proposal
+
+### Cloudwatch streams and groups
+
+[Cloudwatch][concepts] defines *log groups* and *log streams*. To paraphrase the documentation:
+
+>  A log stream is a sequence of log events that share the same source ... For example, an Apache access log on a specific host.
+
+>  Log groups define groups of log streams that share the same retention, monitoring, and access control settings ... For example, if you have a separate log stream for the Apache access logs from each host, you could group those log streams into a single log group called MyWebsite.com/Apache/access_log.
+
+In other words a *log stream* corresponds to the smallest log-producing units.
+A *log group* is a collection of related *log streams*.
+
+We must consider *container logs*, and *node logs*.
+- "application" logs are always container logs
+- "infrastructure" logs are a mix of container and node logs
+- "audit" logs are all node logs.
+
+For *container-scoped* logs we auto-create a log stream for *each container*, using the container UUID as the log stream name.
+- The log stream name is just a unique identifier for the source container.
+- Log *entries* include meta-data (container name, namespace etc.) for the user to search/index.
+  The log stream name is *not* needed for indexing or searching logs.
+- Log streams are for a *single source*, we cannot send logs from multiple nodes on the same log stream.
+
+For *node-scoped* logs we will create "audit"" and "infrastructure" streams for each node: e.g. `node-id/audit`.
+
+Log streams can be grouped by
+
+- **category**: there are three log groups "application", "infrastructure" and "audit".
+- **namespace name**: group per namespace *name*.
+  Used when successive namespace objects with the same name are considered "equivalent".
+  This is a common case, many core k8s tools and APIs work this way.
+- **namespace uuid**: group per namespace *object* using the UUID.
+  Destroying then creating a namespace object with the same name results in a *new log group*.  Use when it is important to distinguish logs from successive namespaces instances with the same name. For example, when namespace re-creation is considered a security risk.
+
+### API fields
+
+New API fields in the `output.cloudwatch` section:
+
+- `region`: (string) AWS region name, required to connect.
+- `retentionDays`: (number) Number of days to keep logs. Defaults to cloudwatch default retention.
```

Comment:
Per conversation with @sichvoge we really would prefer to not add this to LF as it is a variance from other types.  I think TBD

Reply:
Why would the log forwarder participate in setting the retention?

Reply:
I still think we should not expose retention in our API at least. How about we set a default value in our Fluentd for now (e.g. 3d), document that and basically point customers to https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/Working-with-log-groups-and-streams.html#SettingLogRetention so that they can change the retention themselves if they need something different?

That at least doesn't require us to bring something into our API that most people feel not comfortable and if we have to expose it later, fine.